src/main/java/uk/nhs/ctp/transform/QuestionnaireOptionValueTransformer.java

```diff
@@ -0,0 +1,65 @@
+package uk.nhs.ctp.transform;
+
+import java.util.function.Function;
+import org.apache.commons.collections4.Transformer;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.PrimitiveType;
+import org.hl7.fhir.dstu3.model.Questionnaire.QuestionnaireItemOptionComponent;
+import org.hl7.fhir.exceptions.FHIRException;
+import org.springframework.stereotype.Component;
+import uk.nhs.ctp.service.dto.TriageExtension;
+import uk.nhs.ctp.service.dto.TriageOption;
+
+@Component
+public class QuestionnaireOptionValueTransformer
+    implements Transformer<QuestionnaireItemOptionComponent, TriageOption> {
+
+  @Override
+  public TriageOption transform(QuestionnaireItemOptionComponent optionComponent) {
+    TriageOption option = new TriageOption();
+
+    if (optionComponent.hasExtension()) {
+      Extension extension = optionComponent.getExtensionFirstRep();
+      TriageExtension triageExtension = new TriageExtension(extension.getUrl(),
+          extension.getValue().primitiveValue());
+      option.setExtension(triageExtension);
+    }
+
+    if (optionComponent.hasValueCoding()) {
+      Coding optionCode = optionComponent.getValueCoding();
+      option.setCode(optionCode.getCode());
+      option.setDisplay(optionCode.getDisplay());
+      option.setSystem(optionCode.getSystem());
+      return option;
+    }
+
+    //non-code types are mangled into coding-like objects for display on the UI
+    else if (optionComponent.hasValueStringType()) {
```

Comment:
Very minor thing - these can just be `ifs` without the `else`, as the previous `returns` make them completely different code paths